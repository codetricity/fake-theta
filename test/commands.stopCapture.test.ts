import { request, toMatchInvalidHeaderParameterError } from './helper';

describe('POST /osc/commands/execute stopCapture', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.stopCapture' })
      .expect(200);
    expect(res.body).toMatchSnapshot({
      results: {
        fileUrls: [
          expect.stringMatching(/^(http|https):\/\/.+\/.+\/.+\/.+\.JPG/),
        ],
      },
    });
  });
});

describe('THETA X POST /osc/commands/execute stopCapture', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.stopCapture' })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body).toMatchSnapshot({
      results: {
        fileUrls: [
          expect.stringMatching(/^(http|https):\/\/.+\/.+\/.+\/.+\.JPG/),
        ],
      },
    });
  });
});

describe('THETA Z1 POST /osc/commands/execute stopCapture', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.stopCapture' })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body).toMatchSnapshot({
      results: {
        fileUrls: [
          expect.stringMatching(/^(http|https):\/\/.+\/.+\/.+\/.+\/.+\.JPG/),
        ],
      },
    });
  });
});

describe('Unknown Type POST /osc/commands/execute stopCapture', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/execute')
      .send({ name: 'camera.stopCapture' })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
