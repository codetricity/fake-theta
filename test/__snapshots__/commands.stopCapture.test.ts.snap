// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/execute stopCapture should respond with results 1`] = `
{
  "name": "camera.stopCapture",
  "results": {
    "fileUrls": [
      StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
    ],
  },
  "state": "done",
}
`;

exports[`THETA X POST /osc/commands/execute stopCapture should respond with results 1`] = `
{
  "name": "camera.stopCapture",
  "results": {
    "fileUrls": [
      StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
    ],
  },
  "state": "done",
}
`;

exports[`THETA Z1 POST /osc/commands/execute stopCapture should respond with results 1`] = `
{
  "name": "camera.stopCapture",
  "results": {
    "fileUrls": [
      StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
    ],
  },
  "state": "done",
}
`;
